draft-ietf-jsonpath-base.md

```diff
@@ -1176,6 +1181,72 @@ The semantics of regular expressions are as defined in {{-iregexp}}.
 The logical AND, OR, and NOT operators have the normal semantics of Boolean algebra and
 obey its laws (see, for example, {{BOOLEAN-LAWS}}).
 
+##### Function Extensions {#fnex}
+{: unnumbered}
+
+[^unnumbered-bad]
+
+[^unnumbered-bad]: This should not be an unnumbered section, which it is
+    forced to be by the parent section being unnumbered too.
+    The overall structure where the parts of this subsection go to
+    needs to be decided; one part already is in the IANA
+    Considerations in {{iana-fnex}}
+{:source=" -- cabo"}
+
+Beyond the filter expression functionality defined in the preceding
+subsections, JSONPath defines an extension point that can be used to
+add filter expression functionality: "Function Extensions".
+
+A function extension defines a registered name (see {{iana-fnex}}) that
+can be applied to a sequence of zero or more arguments, resulting in a result.
+A function-expression stands for ("returns") a nodelist or a value.
+The arguments are filter expressions that are to be interpreted as
+nodelists or values.
+For each argument and for the result, the function extension defines
+the "kind" of the item, i.e., whether the item is a value or a
+nodelist ("nodes").
+
+~~~ abnf
+function-name           = function-name-first *function-name-char
+function-name-first     = LCALPHA
+function-name-char      = DIGIT / name-first / "_"
+LCALPHA                 = %x61-7A  ; "a".."z"
+
+
+function-expression     = function-name "(" [*(function-argument "," S) function-argument] ")"
```

Comment:
Do we want to allow more whitespace, e.g. just inside the parentheses?

Reply:
space inside parentheses is evil.  Except for line breaks...

Reply:
Well, the answer is yes; we should move any attempt to reduce variability in spacing to its own PR.